### PR TITLE
[FIX] account_peppol: limit demo mode to peppol

### PR DESCRIPTION
**Pull request by aliyatastemirova**
Currently, if account_peppol module is installed on a non-verified database, it is not possible to register on `l10n_it_edi`,
even on the test server. That happens because we do not limit the demo mode to peppol.
This commit adds an additional check for the proxy_type when handling demo.
We wrap `make_request` in the decorator because it's used in many places in account_peppol,
so the condition has to be added inside the decorator.

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aliyatastemirova** [comment]
no task, reported by PGI

**Comment by lordkrandel** [comment]
From what I see, by adding the check on the user type it should solve the problem I had.
So it LGTM. No idea of the flow, though.

(I also have to note that using a decorator that has 3 different cases depending on the model it is called on handled by lambdas... sounds like complexity for nothing. Please remember to keep it simple.)

File: addons/account_peppol/tools/demo_utils.py [hunk 1]
```diff
@@ -143,8 +143,11 @@ def handle_demo(func, self, *args, **kwargs):
     """
     def get_demo_mode_account_edi_proxy_client_user(self, args, kwargs):
         if self.id:
-            return self.edi_mode == 'demo'
-        return self.env['ir.config_parameter'].get_param('account_peppol.edi.mode') == 'demo'
+            return self.edi_mode == 'demo' and self.proxy_type == 'peppol'
+        demo_param = self.env['ir.config_parameter'].get_param('account_peppol.edi.mode') == 'demo'
+        if args:
+            return demo_param and args[1]['proxy_type'] == 'peppol'
```
Comment (smetl) [inline addons/account_peppol/tools/demo_utils.py:149]: That part is not very readable what `args` contains...
Here, this is mocking the method `def _make_request(self, url, params=False):`
So, `params` could be `False`.
So, if `args[1]` could be `False`.
So `args[1]['proxy_type']` could raise a traceback...?
Tell me if I wrongly understand that part :see_no_evil:

Comment (aliyatastemirova) [reply addons/account_peppol/tools/demo_utils.py]: you're absolutely right :see_no_evil: although it shouldn't really happen that we get to the point, where there is no edi user and no params are provided :slightly_smiling_face: 
fixed

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**